File: booking-service/src/main/java/com/example/bookingservice/domain/model/input/CancelReservationInput.java
package com.example.bookingservice.domain.model.input;

import com.example.bookingservice.integration.service.dto.request.CancelReservationClientRequest;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CancelReservationInput {
    private String hotelHoldId;
    private String language;

    public CancelReservationClientRequest toRequest() {
        return CancelReservationClientRequest.builder()
                .hotelHoldId(hotelHoldId)
                .build();
    }

}

File: hotel-service/hotel-service.APPLICATION/model/input/CancelReservationInput.cs
namespace hotel_service.API.dto.request;

public class CancelReservationInput
{
    public string? HotelHoldId { get; set; }

    public CancelReservationInput(string? hotelHoldId)
    {
        HotelHoldId = hotelHoldId;
    }
    
}